### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by maxkfranz**
**General information**

This PR fixes CI so that everything is green.

Associated issues: 

- #140
- #118
- #139

**Checklist**

Author:

- [x] One or more reviewers have been assigned.
- [x] Automated tests have been included in this pull request, if possible, for the new feature(s) or bug fix.
- [x] The associated GitHub issues are included (above).
- [x] Notes have been included (below).

Reviewers:

- [x] All automated checks are passing (green check next to latest commit).
- [x] At least one reviewer has signed off on the pull request.  Reviewers have two business days to review the pull request, after which the author may merge in the pull request unilaterally.

**Notes**

Broad summary (details in commits)

- Fixes CI w.r.t.
  - tests
  - installation
- Remove support for TravisCI in favour of GitHub Actions.  Travis doesn't look like a good longterm solution going forward.

### CODE REVIEW

File: test/root.js [hunk 1]
```diff
@@ -17,4 +17,6 @@ before(async () => {
 after(async () => {
   await expressServer.close();
   console.log("stopped express server");
+
+  // TODO stop cytosnap instance(s)
```
Comment (maxkfranz) [inline test/root.js:21]: @mikekucera, is there a simple way to do this now?  If not, let's leave this TODO for later.

Comment (maxkfranz) [reply test/root.js:21]: OK, it sounds like we should just rely on `--exit` for now.

If there's crash, then we should make an issue for that to investigate/prioritise it.  It sounds like something like that should be resolved before those features make it to master.  Maybe there's a memory issue or something simple.  We'll have to investigate and see

**Review by mikekucera — APPROVED** [review]

**Comment by d2fong** [comment]
One of the tests fails for me still:
```
  1) History (Snapshots)
       creates thumbnails:
     Error: Timeout of 15000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/dylan/src/idekerlab/cytoscape-explore/test/history-test.js)
      at listOnTimeout (node:internal/timers:557:17)
      at processTimers (node:internal/timers:500:7)
```

**Comment by d2fong** [comment]
But I ran it again and it passes.  Maybe the time limit should be increased?

**Comment by maxkfranz** [comment]
15s is too long. If it takes that long, the user is going to be left hanging.  Ultimately still a test failure in my opinion. It’s hard to get puppeteer setups just right

We could comment out that test and make an issue for the problem if we want to move forward quickly

File: test/root.js (hunk 1 above)
Comment (maxkfranz) [reply test/root.js:21]: See #143

**Comment by maxkfranz** [comment]
The test is commented out in [608f827](https://github.com/cytoscape/cytoscape-explore/pull/141/commits/608f827975b45b47b8af4cd0debc6e6ede8de17a).  This PR could be merged in, but #145 needs to be fixed before dockerised CI instances will run correctly

**Comment by maxkfranz** [comment]
@d2fong, do you want to put the patch for #145 in this PR?  It looks like GH Actions was able to reproduce the checksum error affecting docker: https://github.com/cytoscape/cytoscape-explore/runs/5452877547?check_suite_focus=true

**Comment by d2fong** [comment]
Ok I pushed a fix.  We were working on refactoring the monorepo stuff so the dependency broke.  Sorry.

**Comment by d2fong** [comment]
@keiono, how would I update the ndex-client dependency using the new monorepo?

**Comment by maxkfranz** [comment]
I think you can use a git repo as a dependency, but the package manager is strict with the lock files now (for security etc.).  It looks like there's a hash check that's failing.  So, we need the hash to be the same as what's installed when `npm install` was used to add to `package.json`.

How we make the hash consistent is up to you guys.  Here are some possibilities:

(1) Use a stable branch in your repo that's not changing.  Every time you change it, you have to update the lock file here in CE.  I'm not crazy about this option, since it's easy to get things out of sync and the failures can be non-obvious.

(2) Point to a specific commit in the git address instead of a branch.

(3) Publish to npm so version numbers can be used.  Maybe a beta version like `1.0.0-beta1`?

Maybe you've got a better/simpler suggestion to get the hash right than these three?  Whatever works

**Comment by d2fong** [comment]
I think the issue might be related to docker and git not having the right permissions.  
```
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/ndexbio/ndex-js-client.git
npm ERR! 
npm ERR! No user exists for uid 1001
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 12[8](https://github.com/cytoscape/cytoscape-explore/runs/5453467076?check_suite_focus=true#step:6:8)
```

**Comment by maxkfranz** [comment]
Could we use a public https address instead of a private ssh one?  Is ssh needed if we're only cloning?

**Comment by d2fong** [comment]
Ok Kei and Jing updated the proper version/monorepo

**Comment by maxkfranz** [comment]
@jingjingbic, @d2fong, @chrtannus -- you can merge `dev` into your branches for your PRs now to help make your PRs green

**Comment by chrtannus** [comment]
Great! I've just merged `dev` into `feature/recent-networks` and the [PR](https://github.com/cytoscape/cytoscape-explore/pull/142) and it's all green now.

**Pull request merged**